examples/text-generation/generation.py

```diff
@@ -0,0 +1,91 @@
+import argparse
+import time
+
+import torch
+from transformers import AutoTokenizer, set_seed
+
+from optimum.neuron import NeuronModelForCausalLM
+
+
+def generate(model, tokenizer, prompts, length, temperature):
+    # Specifiy padding options for decoder-only architecture
+    tokenizer.pad_token_id = tokenizer.eos_token_id
+    tokenizer.padding_side = "left"
+
+    # Encode tokens and generate using temperature
+    tokens = tokenizer(prompts, return_tensors="pt", padding=True)
+    start = time.time()
+    with torch.inference_mode():
+        sample_output = model.generate(
+            **tokens, do_sample=True, max_length=length, temperature=temperature, top_k=50, top_p=0.9
+        )
+    end = time.time()
+    outputs = [tokenizer.decode(tok) for tok in sample_output]
+    return outputs, (end - start)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(help="Action to perform", dest="action")
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument("model", type=str, help="The HF Hub model id or a local directory.")
+    compile_parser = subparsers.add_parser("compile", parents=[parent_parser], help="Convert model to Neuron.")
+    compile_parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=1,
+        help="The batch size.",
+    )
+    compile_parser.add_argument(
+        "--num_cores", type=int, default=1, help="The number of cores on which the model should be split."
+    )
+    compile_parser.add_argument(
+        "--auto_cast_type", type=str, default="f32", choices=["f32", "f16", "bf16"], help="One of f32, f16, bf16."
+    )
+    compile_parser.add_argument(
+        "--save_dir", type=str, help="The save directory. Allows to avoid recompiling the model every time."
+    )
+    run_parser = subparsers.add_parser(
+        "run", parents=[parent_parser], help="Generate tokens using the specified model."
+    )
+    run_parser.add_argument(
+        "--prompts",
+        type=str,
+        default="One of my fondest memory is",
+        help="The prompts to use for generation, using | as separator.",
+    )
+    run_parser.add_argument("--length", type=int, default=128, help="The number of tokens in the generated sequences.")
+    run_parser.add_argument(
+        "--temperature",
+        type=float,
+        default=1.0,
+        help="The temperature to generate. 1.0 has no effect, lower tend toward greedy sampling.",
+    )
+    run_parser.add_argument("--seed", type=int, default=None, help="Pass a seed for reproducibility.")
+    args = parser.parse_args()
+    if args.action == "compile":
```

Comment:
I wonder if we shall use `export` instead of `compile` here. Back to the moment we chose `export` other than `compile`, and it's kinda nice to keep the example aligned with the API naming.

Reply:
I hesitated a lot on that one. Will call it export.